scripts/setup.sh

```
@@ -0,0 +1,956 @@
+}
+
+function check_required_commands() {
+    local commands=("git" "yarn" "go")  
+    for cmd in "${commands[@]}"; do
+        check_command "$cmd"
+    done
+}
+
+# check if the go version is installed 
+function check_go_version() {
+    local go_version=$(go version | awk '{print $3}' | sed 's/^go//')
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)